frontend/src/metabase/query_builder/actions.js

```diff
@@ -154,6 +158,22 @@ export const popState = createThunkAction(
   POP_STATE,
   location => async (dispatch, getState) => {
     dispatch(cancelQuery());
+
+    const zoomedObjectId = getZoomedObjectId(getState());
+    if (zoomedObjectId) {
+      const { locationBeforeTransitions } = getState().routing;
+
+      const previouslyZoomedObjectId =
+        locationBeforeTransitions?.state?.objectId;
+
+      if (previouslyZoomedObjectId) {
```

Comment:
For some reason, if you zoom on object 1, go to object 2, go back to the previous, and then go back to question — it keeps `previouslyZoomedObjectId` as object 1 and you can't return to your question. Do you know what can be the reason?